### PR TITLE
vdaf: Have prepare_init() output the first-round prep message

### PR DESCRIPTION
**Pull request by cjpatton**
Modifies the Aggregator API so that `prepare_init()` outputs the first
round Prepare message. This is less general, since it excludes VDAFs
that may have 0 rounds of communication during the Prepare phase. The
upside is that it eliminates a number of rough edges in both the API and
the implementation. For example, there is never a need to serialize the
initial state prior to producing the first-round message.

FWIW, I think this change is useful regardless of if/how we resolve https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/60.

### CODE REVIEW

**Comment by tgeoghegan** [comment]
I haven't reviewed these changes yet, but IIUC this is part of implementing vdaf-01, so we should do something like what is described in https://github.com/divviup/libprio-rs/issues/216 before we merge this to main.

**Comment by cjpatton** [comment]
I don't think it's strictly related to changes in -01, since we're not actually changing the crypto. However I'm fine with holding off on merging.

**Review by BranLwyd — APPROVED** [review]
I like this change! It fits well with how Janus implements things currently, and smooths off (what I understand to be) a rough-edge of the API.

**Review by tgeoghegan — APPROVED** [review]
I agree that this change is good, I'm just worried that we will end up doing our interop tests with an implementation of an unclear mix of `vdaf-00` and `vdaf-01` which may make it harder to debug problems. So that's why I want to be explicit about whether we are taking this change into an implementation of `vdaf-00` or `-01`.

**Comment by cjpatton** [comment]
Ack, @tgeoghegan -- I'm happy to hold off on merging until we're closer to -01.

File: src/vdaf/prio3.rs
Comment (cjpatton) [inline src/vdaf/prio3.rs:611]: Remove this comment, as it's no longer relevant.

Comment (cjpatton) [reply src/vdaf/prio3.rs]: Done (similarly above).

File: src/vdaf.rs
Comment (tgeoghegan) [inline src/vdaf.rs:433]:
```suggestion
            .expect("failed to decode prepare message")
```
But in any case, why not propagate the error? I think something like
```
let mut inbound = vdaf.prepare_preprocess(outbound.iter().map(|encoded| V::PrepareMessage::get_decoded_with_param(...)).transpose().collect::<Vec<Result<_, _>>>())
```
might work.

File: src/vdaf.rs
Comment (tgeoghegan) [inline src/vdaf.rs:458]:
```suggestion
                    .expect("failed to decode prepare message")
```
But as above, would prefer to propagate the error instead of crashing.

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**
